src/cdomains/intDomain.ml

```diff
@@ -923,6 +923,461 @@ struct
+    | [] -> failwith "unsupported: equal_to with bottom"
+    | [`Eq] ->  `Eq 
+    | ys -> if List.for_all (fun x -> x = `Neq) ys  then `Neq else `Top  
+     
+  include Std (struct type nonrec t = t let name = name let top_of = top_of let bot_of = bot_of let show = show let equal = equal end)
+
+  let set_overflow_flag ~cast ~underflow ~overflow ik =
+    let signed = Cil.isSigned ik in
+    if !GU.postsolving && signed && not cast then
+      Goblintutil.svcomp_may_overflow := true;
+    let sign = if signed then "Signed" else "Unsigned" in
+    match underflow, overflow with
+    | true, true ->
+      M.warn ~category:M.Category.Integer.overflow ~tags:[CWE 190; CWE 191] "%s integer overflow and underflow" sign
+    | true, false ->
+      M.warn ~category:M.Category.Integer.overflow ~tags:[CWE 191] "%s integer underflow" sign
+    | false, true ->
+      M.warn ~category:M.Category.Integer.overflow ~tags:[CWE 190] "%s integer overflow" sign
+    | false, false -> assert false
+  
+  let norm ?(cast=false) ik = function 
+  | None -> [] 
+  | Some (x,y) ->
+    if Ints_t.compare x y > 0 then 
+      []
+    else begin
+      let (min_ik, max_ik) = range ik in
+      let underflow = Ints_t.compare min_ik x > 0 in
+      let overflow = Ints_t.compare max_ik y < 0 in
+      if underflow || overflow then
+        begin
+          set_overflow_flag ~cast ~underflow ~overflow ik;
+          if should_wrap ik then (* could add [|| cast], but that's GCC implementation-defined behavior: https://gcc.gnu.org/onlinedocs/gcc/Integers-implementation.html#Integers-implementation *)
+            (* We can only soundly wrap if at most one overflow occurred, otherwise the minimal and maximal values of the interval *)
+            (* on Z will not safely contain the minimal and maximal elements after the cast *)
+            let diff = Ints_t.abs (Ints_t.sub max_ik min_ik) in
+            let resdiff = Ints_t.abs (Ints_t.sub y x) in
+            if Ints_t.compare resdiff diff > 0 then
+              [range ik]
+            else
+              let l = Ints_t.of_bigint @@ Size.cast ik (Ints_t.to_bigint x) in
+              let u = Ints_t.of_bigint @@ Size.cast ik (Ints_t.to_bigint y) in
+              if Ints_t.compare l u <= 0 then
+                [(l, u)]
+              else
+                (* Interval that wraps around (begins to the right of its end). We CAN represent such intervals *)
+                [(min_ik, u); (l, max_ik)]
+          else if not cast && should_ignore_overflow ik then
+            let tl, tu = range ik in
+            [Ints_t.max tl x, Ints_t.min tu y]
+          else
+            [range ik]
+        end
+      else 
+        [(x,y)]
+    end
+
+  let leq (xs: t) (ys: t) = match xs, ys with
+    | [], _ -> true
+    | _, [] -> false
+    | _::_, _::_ -> let leq_interval = fun (al, au) (bl, bu) -> Ints_t.compare al bl >= 0 && Ints_t.compare au bu <= 0 in
+      List.for_all (fun x -> List.exists (fun y -> leq_interval x y) ys) xs   
```

Comment:
This looks overly expensive, I don't think this operation needs to be quadratic!